src/llama.cpp

```diff
@@ -18733,7 +18733,7 @@ struct llama_model_params llama_model_default_params() {
         /*.check_tensors               =*/ false,
     };
 
-#ifdef GGML_USE_METAL
+#if defined(GGML_USE_METAL) || defined(GGML_USE_MUSA)
```

Comment:
I don't think this is necessary. If MUSA works the same way as HIP then CPU+GPU hybrid inference should work without the need for any additional changes.

Reply:
Yes. This is quite weird. By using Ollama, the model could be offloaded to the GPU. While using llama-server directly, I saw this log: `llm_load_tensors: offloaded 0/33 layers to GPU`. So I marked `result.n_gpu_layers = 999` the same as metal.

Reply:
This makes sense with Metal because it uses the same memory as the CPU, so there is rarely any reason to not fully offload the model. Problems with ollama should be fixed in ollama.

Reply:
>While using llama-server directly, I saw this log: llm_load_tensors: offloaded 0/33 layers to GPU. 

Did you set the `--n-gpu-layers` CLI argument?

Reply:
No. I just run llama-server like:

```bash
root@8cb6823fed5c:/ws# ./llama-server
INFO [                    main] build info | tid="139686874033984" timestamp=1720524760 build=3331 commit="777add8b"
INFO [                    main] system info | tid="139686874033984" timestamp=1720524760 n_threads=6 n_threads_batch=-1 total_threads=12 system_info="AVX = 1 | AVX_VNNI = 1 | AVX2 = 1 | AVX512 = 0 | AVX512_VBMI = 0 | AVX512_VNNI = 0 | AVX512_BF16 = 0 | FMA = 1 | NEON = 0 | SVE = 0 | ARM_FMA = 0 | F16C = 1 | FP16_VA = 0 | WASM_SIMD = 0 | BLAS = 1 | SSE3 = 1 | SSSE3 = 1 | VSX = 0 | MATMUL_INT8 = 0 | LLAMAFILE = 1 | "
llama_model_loader: loaded meta data with 23 key-value pairs and 201 tensors from models/7B/ggml-model-f16.gguf (version GGUF V3 (latest))
...
ggml_cuda_init: found 1 MUSA devices:
  Device 0: MTT S4000, compute capability 2.2, VMM: yes
llm_load_tensors: ggml ctx size =    0.27 MiB
llm_load_tensors: offloading 0 repeating layers to GPU
llm_load_tensors: offloading non-repeating layers to GPU
llm_load_tensors: offloaded 0/33 layers to GPU
...
```

Reply:
The default value for `--n-gpu-layers` is 0 so you are effectively not using the GPU at all.

Reply:
Thanks for letting me know this. I will revert this change.

Reply:
Please see the latest commits. Thanks.